### PR TITLE
Move StreamK and StreamD folder up

### PR DESCRIPTION
**Pull request by rnjtranjan**

### CODE REVIEW

**Comment by rnjtranjan** [comment]
Fixes:https://github.com/composewell/streamly/issues/2477

**Comment by rnjtranjan** [comment]
Fixes:https://github.com/composewell/streamly/issues/1666

File: core/src/Streamly/Data/Stream.hs [hunk 1]
```diff
@@ -117,7 +117,7 @@ module Streamly.Data.Stream
     -- | Functions ending in the general shape @Stream m a -> m b@ or @Stream m
     -- a -> m (b, Stream m a)@
     --
-    -- See also: "Streamly.Internal.Data.Stream.StreamD.Eliminate" for @Pre-release@
+    -- See also: "Streamly.Internal.Data.Stream.Eliminate" for @Pre-release@
```
Comment (harendra-kumar) [inline core/src/Streamly/Data/Stream.hs:120]: This is hidden.

File: core/src/Streamly/Data/Stream.hs [hunk 2]
```diff
@@ -259,7 +259,7 @@ module Streamly.Data.Stream
     -- * Scanning
     -- | Stateful one-to-one transformations.
     --
-    -- See also: "Streamly.Internal.Data.Stream.StreamD.Transform" for
+    -- See also: "Streamly.Internal.Data.Stream.Transform" for
```
Comment (harendra-kumar) [inline core/src/Streamly/Data/Stream.hs:262]: This is hidden.

File: core/src/Streamly/Data/Stream.hs [hunk 3]
```diff
@@ -485,7 +485,7 @@ module Streamly.Data.Stream
     -- For example, instead of calling them on a stream of chars call them on a
     -- stream of arrays before flattening it to a stream of chars.
     --
-    -- See also: "Streamly.Internal.Data.Stream.StreamD.Exception" for
+    -- See also: "Streamly.Internal.Data.Stream.Exception" for
```
Comment (harendra-kumar) [inline core/src/Streamly/Data/Stream.hs:488]: This is hidden.

File: core/streamly-core.cabal [hunk 4]
```diff
@@ -309,9 +309,8 @@ library
                      -- streamly-core-streams
                      , Streamly.Internal.Data.StreamK
                      -- StreamD depends on streamly-array-types
-                     , Streamly.Internal.Data.Stream.StreamD
-                     , Streamly.Internal.Data.Stream.Common
                      , Streamly.Internal.Data.Stream
+                     , Streamly.Internal.Data.Stream.Common
```
Comment (harendra-kumar) [inline core/streamly-core.cabal:313]: Modules are listed in dependency order, lower level modules come before higher level ones. This should be before Data.Stream.

**Review by harendra-kumar — CHANGES_REQUESTED** [review]
We should deprecate this core/src/Streamly/Internal/Data/Stream/StreamD.hs rather than deleting it - in case someone is importing from it.

**Review by harendra-kumar — APPROVED** [review]

**Pull request merged**